modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -348,15 +360,20 @@ public void visit(BLangIf ifStmt) {
             ifStmt.elseStmt.accept(this);
             this.statementReturns = ifStmtReturns && this.statementReturns;
         }
+        this.retryStmtCheckStack.pop();
     }
 
     @Override
     public void visit(BLangWhile whileNode) {
+        this.retryStmtCheckStack.push(false);
```

Comment:
Is "if" and "while" the only places we have to mark this, what about inside another transaction block? ..

Reply:
Fixed with f6fe91e29e529078b0c173726ea698ee63e9fe62 and 8b9b94c1e3f1ec5f1e497fffadee03de43b284fa.Thanks.